packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1778,6 +1839,10 @@ export function getAllModelsWithDiagnostics(
       diagnostics.pipe(updateTypesFromOperation(context, operation));
     }
     const ogs = listOperationGroups(context, client);
+    const clientInitModel = getClientInitialization(context, client.type);
+    if (clientInitModel) {
+      getSdkInitializationType(context, client, clientInitModel);
+    }
```

Comment:
is it a must that we need to get the client initialization model before generating sdk pacakge thing?
also, seems this logic forget to deal with operation group.

Reply:
If we don't have it here, I'm not able to pick up the model. I'll add logic for operation groups too